### DISCUSSION TITLE
Introduce a second test environment

### DISCUSSION THREAD

**Discussion by danrot** [symfony/symfony#48157] [Q&A]
I am working on a project that has a "legacy" test setup I want to replace with one that works solely with PHPUnit. Now I don't want to replace everything at once, so I came up with the idea of introduce a new environment `test_phpunit` to run the new set of tests (I need that because I also want to handle some test services differently).

So what I am doing is executing the `phpunit` command with the environment variable set to `APP_ENV=test_phpunit`. I have also created the same files as for the test environment `.env.test_phpunit`, `config/packages/test_phpunit/config.yaml` and `config/services_test_phpunit.yaml`.

Unfortunately this does not seem to start the framework in `test` mode, i.e. I have to set the `framework.test` option manually in `config/packages/test_phpunit/config.yaml`, which is NOT necessary for the `test` environment. However, after doing that I get another error claiming the session cannot be started because something has already been sent, so I suspect that the cleaning of the output buffer also seems to work differently.

Now my question is: Is having multiple different testing environments possible? Or does the `test` envionment enjoy some special handling that makes test work? Or is there just something else I am currently missing?

**Comment by stof** [comment]
> Unfortunately this does not seem to start the framework in `test` mode, i.e. I have to set the `framework.test` option manually in `config/packages/test_phpunit/config.yaml`, which is NOT necessary for the `test` environment.

This is actually needed also for the `test` environment. It is just that the recipe for FrameworkBundle contains that config by default (and so if you create a project with Flex, you get it automatically). See https://github.com/symfony/recipes/blob/833082a0826a7041534370a787b40d5dc0d86fc4/symfony/framework-bundle/6.2/config/packages/framework.yaml#L21-L23 for the corresponding config (or https://github.com/symfony/recipes/blob/833082a0826a7041534370a787b40d5dc0d86fc4/symfony/framework-bundle/5.1/config/packages/test/framework.yaml#L1-L2 for older Symfony versions that don't support the `when@test` feature in Yaml config files).

**Reply by danrot** [reply]
I am using Pimcore here, and this does not seem to be a Symfony Flex project. I also don't find any line containing `test: true` in the entire project...

Also, if I am executing `APP_ENV=test ddev console debug:config framework test` I only get the following error message:

```
[ERROR] Unable to find configuration for "framework.test".
```

AFAIK this should mean that the configuration is not set at all, right? Still, the tests are working in this environment, while they are failing with a message saying that I should set the `framework.test` configuration in the other case. Are there any other options to set this configuration that cannot be exposed using `debug:config`?
